runtime/compiler/env/ProcessorDetection.cpp

```diff
@@ -619,6 +619,9 @@ TR_J9VM::initializeProcessorType()
          processorDescription.processor = OMR_PROCESSOR_PPC_7XX;
 
       TR::Compiler->target.cpu = TR::CPU::customize(processorDescription);
+      J9JavaVM *javaVM = _jitConfig->javaVM;
+      if (!J9_ARE_ANY_BITS_SET(javaVM->extendedRuntimeFlags2, J9_EXTENDED_RUNTIME2_ENABLE_PORTABLE_SHARED_CACHE))
+         TR::Compiler->relocatableTarget.cpu = TR::CPU::customize(processorDescription);
```

Comment:
Wondering why relocatableTarget.cpu is not set on other platforms ...
they need to handle portable SCC too.

Reply:
at line 665,

    _jitConfig->relocatableTargetProcessor = TR::Compiler->relocatableTarget.cpu.getProcessorDescription();

that looks like ending up with garbage relocatableTargetProcessor if relocatbleTarget.cpu is not set.  ???

Reply:
https://github.com/AlenBadel/omr/blob/a0dc9ace85fc6298d06db9c5a979d90f720b0576/compiler/env/OMRCompilerEnv.cpp#L167

With non-portable AOT compilation, the RelocatableTarget is initialized to the CPU that is running on the machine. On Power, it needs to be overwritten based on the presence of the ENV var.

To my knowledge `initializeRelocatableTargetEnvironment` occurs before `initializeProcessorType` so the relocatable target's cpu is always initalized.

Reply:
It looks like very early on we set the relocatableTarget.cpu to whatever is running on the machine via `relocatableTarget.cpu = TR::CPU::detect(TR::Compiler->omrPortLib);`, same as what happens for `target.cpu`. 

However, unlike `target.cpu`, `relocatableTarget.cpu` gets updated later on depending on whether it's an AOT compile or load run. If it's an AOT compile run, we'll do it in `TR_SharedCacheRelocationRuntime::createAOTHeader` right before the very first compile; if it's a load run we do it in `aboutToBootstrap`.

It might be worth adding your change to all three of the `TR::CPU::customize(processorDescription)` calls in that file since if we're not in portable SCC mode, the `relocatableTarget.cpu` should be the same as the `target.cpu`.

Reply:
@dsouzai   this change is only temporary for Alen testing purposes.  Eventually (AOT and codecache alignment issues fixed), this environment variable is going to be eliminated. Returning POWER10 to its own state, instead of faking as a POWER9.  So, this PR is no need to be merged.